base_classes/nyaml/NXbeam.yaml

```diff
@@ -0,0 +1,258 @@
+doc: | 
```

Comment:
For each docstring we should have at least the optional possibility but a field and keyword/command for this realized whereby we can add a reference to a concept of another ontology (OBO, BFO, EMMO) ...

Also the docstrings should really be inspected and cleaned from instructing to humans what they should format there, if we feel we need to express how things should be formatted behind a field we talk about constraint management

Reply:
Yes, I support this. We could add something like `ontology_reference` or `ontology_uri` as sub-field of `doc`?

I like the idea of adding constraints (I'm thinking of describing formulas etc. properly here as well) but I have the feeling that currently its a bit out of our scope of our simple prototype example here.